### DISCUSSION TITLE
Is there a way to broadcast key inputs to multiple panes?

### DISCUSSION THREAD

**Discussion by shaozi** [wezterm/wezterm#1130] [Q&A] [answered]
I want to input same key strokes to multiple panes. i.e. press `a`, all panes receive `a`.

**Comment by wez** [comment, marked as answer]
wezterm doesn't support this today

**Comment by piechologist** [comment]
Since broadcasting was the only feature I was missing in wezterm I'm fiddling with a key binding. I just compose a command in one pane and with SUPER-Enter copy that command to the other panes in the same tab. That's not a full-fledged solution as in iTerm but all I need. It might be useful for somebody else.

Caveat: for now, I'm using a dirty trick to grab the command line buffer. I get the whole viewport and cut the beginning off up to the last prompt. So, unfortunately I need a regex to find the prompt. Is there something like `wezterm.action.SelectTextAtMouseCursor 'SemanticZone'` but for the text cursor (insertion point)? Or any other idea how to capture the command buffer?

```lua
 { key = 'Enter', mods = 'CMD', action = wezterm.action_callback(function(win, pane)
    local prompt_regex = '.+\xee\x82\xb0 '  -- my prompt ends with a powerline character 
    local tab = pane:tab()
    local txt = pane:get_logical_lines_as_text()
    local cmd = string.gsub(txt, prompt_regex, '')
    wezterm.log_info('Command:', cmd)
    for _, p in ipairs(tab:panes()) do
      if p:pane_id() == pane:pane_id() then
        wezterm.log_info('Sending \\n to own pane', p)
        p:send_text('\n')
      else
        wezterm.log_info('Sending command\\n to pane', p)
        p:send_text(cmd..'\n')
      end
    end
  end)
},
```

**Reply by shaozi** [reply]
This is nice. But it will be nice to have build in broadcast feature. 
@wez do you have plan to implement this? I bet it is a high demand feature.

**Reply by wez** [reply]
I have no immediate plans to work on an input broadcast feature. This is the tracking issue:
* https://github.com/wez/wezterm/issues/2658

Adding `Pane` methods that allow working with semantic zones has less UI/UX uncertainty than input broadcasting and is something that could get implemented more quickly. @piechologist could you open an enhancement issue for that? Having your feedback on what the API might look like will help.

**Reply by piechologist** [reply]
Thank you for looking into this. I've opened #2968.

That's my first encounter with Lua. Looks like a great choice for scripting an application! It's quite easy to grasp but very powerful and your docs are awesome.

Also, I like doing everything in a config file. It's more flexible than GUI settings are, plus, I can copy the file without struggling with macOS's `defaults` services. 

Cheers!

**Reply by matti** [reply]
this is the only missing feature to switch from iterm2

**Reply by pasquale95** [reply]
Hi,
do we have news about this feature? Is it still pending?

**Reply by whughesiii2187** [reply]
This sounds like a case of creator doesn't use therefore creator will not implement.

**Reply by sshishov** [reply]
@wez @piechologist guys, please note that the issu is closed (due to inactivity) and it is not yet implemented.
